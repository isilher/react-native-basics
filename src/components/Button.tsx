import React, { useEffect, useState } from "react"
import { Pressable, StyleSheet, ViewStyle } from "react-native"

export const MyButton: React.FC<{
  onPress: () => void
  theme: string
}> = ({ onPress, theme, children }) => {
  const [opacity, setOpacity] = useState(1)
  const [color, setColor] = useState("pink")

  useEffect(() => {
    if (theme === "naval") setColor("blue")
    if (theme === "football") setColor("orange")
  }, [theme])

  const buttonStyles = StyleSheet.compose(styles.myButton, {
    opacity,
    backgroundColor: color,
  })

  const onPressIn = () => {
    setOpacity(0.5)
    setColor("red")
  }

  const onPressOut = () => {
    onPress()
    setOpacity(1)
    if (theme === "naval") setColor("blue")
    if (theme === "football") setColor("orange")
  }

  return (
    <Pressable
      style={buttonStyles}
      onPressOut={onPressOut}
      onPressIn={onPressIn}
    >
      {children}
    </Pressable>
  )
}

const styles = StyleSheet.create<{
  myButton: ViewStyle
}>({
  myButton: {
    height: 50,
    width: "50%",
    backgroundColor: "pink",
  },
})
