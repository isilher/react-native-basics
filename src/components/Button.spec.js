import React from "react"
import { render, fireEvent } from "@testing-library/react-native"

import { MyButton } from "./Button"

describe("<MyButton />", () => {
  it("renders correctly", () => {
    const { toJSON } = render(<MyButton title="Test!" onPress={() => {}} />)

    expect(toJSON()).toMatchSnapshot()
  })

  it("triggers the given onPress method", () => {
    const onPressMock = jest.fn()

    const { getByText } = render(
      <MyButton title="Test!" onPress={onPressMock} />
    )
    fireEvent(getByText("Test!"), "onPressOut")

    expect(onPressMock).toHaveBeenCalled()
  })
})
