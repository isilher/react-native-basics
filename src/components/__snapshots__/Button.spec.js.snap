// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 50,
        "justifyContent": "center",
        "paddingHorizontal": 12,
      },
      Object {
        "backgroundColor": "blue",
        "opacity": 1,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 30,
      }
    }
  >
    Test!
  </Text>
</View>
`;
